src/store/TopicStore.ts

```diff
@@ -10,35 +10,54 @@ import { useUserStore } from './UserStore'
 const topicsAPI = new TopicsAPI()
 const topicsAPIv2 = new TopicsAPI({ version: 2 })
 
+type SortCriterions = '-created_at' | '-last_modified' | 'name'
+
 export interface RootState {
   data: Topic[]
   isLoaded: boolean
+  sort: SortCriterions
 }
 
 export const useTopicStore = defineStore('topic', {
   state: (): RootState => ({
     data: [],
     // flag for initial/remote loading of data
-    isLoaded: false
+    isLoaded: false,
+    sort: '-created_at'
   }),
   getters: {
     // Computed property to get topics owned by the current user sorted by last_modified
     userTopics(): ComputedRef<Topic[]> {
       const userStore = useUserStore()
       return computed(() => {
         if (!userStore.isLoggedIn) return []
-        return this.sortedByLastModifiedDesc.filter(
+        return this.sortedByDateDesc('last_modified').filter(
           (topic: Topic) => topic.owner?.id === userStore.data?.id
         )
       })
     },
-    sortedByLastModifiedDesc: (state) => {
-      return [...state.data].sort((a, b) => {
-        const dateA = new Date(a.last_modified)
-        const dateB = new Date(b.last_modified)
-        return dateB.getTime() - dateA.getTime()
-      })
-    }
+    sorted(): Topic[] {
+      switch (this.sort) {
```

Comment:
Ca vaudrait pas le coup d'en faire un enum/des constantes ?
TS doit pouvoir nous éviter les bugs de typos j'imagine ?

Reply:
It is https://github.com/opendatateam/udata-front-kit/pull/391/files#diff-6afc24ec7c6237886350aeed8fc74fc276506d14ad46cc8992c9272b183623cfR13

Reply:
J'avais vu le typage mais j'avais pas poussé la réflexion assez loin : les strings deviennent de fait des symboles. Pas évident à suivre sans IDE mais ça se tient :)